astropy/coordinates/transformations.py

```diff
@@ -70,6 +70,20 @@ def frame_set(self):
 
         return self._cached_frame_set.copy()
 
+    @property
+    def frame_attrnames_set(self):
+        """
+        A `frozenset` of all the attribute names of all frame classes in this `TransformGraph`.
+        """
+        if self._cached_frame_attrnames_set is None:
+            result = set()
+            for frame_cls in self.frame_set:
+                for attr in frame_cls.get_frame_attr_names().keys():
```

Comment:
why not just `result.update(frame_cls.get_frame_attr_names())` instead of the manual loop?